src/features/defi/providers/cosmos/components/CosmosManager/Overview/WithdrawCard.tsx

```diff
@@ -0,0 +1,97 @@
+import { Button, Stack, useColorModeValue } from '@chakra-ui/react'
+import { Asset } from '@shapeshiftoss/asset-service'
+import dayjs from 'dayjs'
+import { DefiParams, DefiQueryParams } from 'features/defi/contexts/DefiManagerProvider/DefiCommon'
+import { useMemo } from 'react'
+import { FaArrowDown } from 'react-icons/fa'
+import { Amount } from 'components/Amount/Amount'
+import { IconCircle } from 'components/IconCircle'
+import { Text } from 'components/Text'
+import { useBrowserRouter } from 'hooks/useBrowserRouter/useBrowserRouter'
+import { bnOrZero } from 'lib/bignumber/bignumber'
+import {
+  selectFirstAccountSpecifierByChainId,
+  selectUnbondingEntriesByAccountSpecifier,
+} from 'state/slices/selectors'
+import { useAppSelector } from 'state/store'
+
+type WithdrawCardProps = {
+  asset: Asset
+}
+
+export const WithdrawCard = ({ asset }: WithdrawCardProps) => {
+  const { query } = useBrowserRouter<DefiQueryParams, DefiParams>()
+  const { contractAddress } = query
+
+  const accountSpecifier = useAppSelector(state =>
+    selectFirstAccountSpecifierByChainId(state, asset?.chainId),
+  )
+
+  const undelegationEntries = useAppSelector(state =>
+    selectUnbondingEntriesByAccountSpecifier(state, {
+      accountSpecifier,
+      validatorAddress: contractAddress,
+      assetId: asset.assetId,
+    }),
+  )
+
+  const hasClaim = useMemo(() => Boolean(undelegationEntries.length), [undelegationEntries])
+  const textColor = useColorModeValue('black', 'white')
+
+  return (
+    <Stack px={8} py={6}>
+      <Text fontWeight='medium' translation='defi.modals.cosmosOverview.withdrawals' />
+      {!hasClaim ? (
+        <Text color='gray.500' translation='defi.modals.cosmosOverview.emptyWithdraws' />
+      ) : (
+        undelegationEntries.map(({ amount, completionTime }) => {
```

Comment:
memoize for perf?

Reply:
Memoized in 86cdaddb4b2bbd34e533fc7d114f93149f10d02d